### ISSUE TITLE
Add option for common anode/cathode

### ISSUE THREAD

**Issue by asergunov** [asergunov/7segment_gpio#8]
Add option for common anode/cathode so user will not need to invert pins one by one

**Comment by I-hate-2FA** [comment]
https://gitlab.com/ailife8881/7segment-gpio

**Comment by asergunov** [comment]
Awesome! You can open PR to contribute to this repo.

**Comment by I-hate-2FA** [comment]
i cant upload code on github, because account might get locked, my other account @AILIFE4798 is locked for login now, because it force you to enable 2fa, so this account will stay for comment only, you can just copy the code from gitlab and upload it here, then i can delete the repo on gitlab

**Comment by asergunov** [comment]
I see. Give me second. I’ll pull your branch.

BTW I think you can use gitlab in your tests if you like.

Ср, 3 июля 2024 г. в 15:56, AILIFE ***@***.***>:

**Comment by I-hate-2FA** [comment]
yes i have used
```yaml
external_components:
  - source:
      type: git
      url: https://gitlab.com/ailife8881/7segment-gpio
      ref: main
    refresh: 0s  
```

**Comment by asergunov** [comment]
I see. Here is your version https://github.com/asergunov/7segment_gpio/tree/contribution2 

Should I open PR or you can?

**Comment by I-hate-2FA** [comment]
it didnt work for me
`Pull request creation failed. Validation failed: must be a collaborator `
maybe you should

**Comment by asergunov** [comment]
I see. https://github.com/asergunov/7segment_gpio/pull/10

**Comment by asergunov** [comment]
I've invited you to collaborators so you should be able to add yours

**Comment by asergunov** [comment]
What's wrong with 2FA BTW? Just curious.

**Comment by I-hate-2FA** [comment]
> What's wrong with 2FA BTW? Just curious.

just not convinient to log in on browser, only accepted 2fa is send code via email, because it can be opened on browser also
you can use 2fa app in android emulator, but what if i need to use my laptop, its just not nice to force your user, for me creating micro controller code security is not a concern
the new account is just named in the hope itll last longer

**Comment by asergunov** [comment]
There was incendent with xz lately. CVE-2024-3094. When ssh was
intentionally compromised. Imagine if it was done from stolen account? I
think 2fa is essential here because it’s not about you but about people who
trust your account history.

Ср, 3 июля 2024 г. в 16:35, AILIFE ***@***.***>:

**Comment by I-hate-2FA** [comment]
i know, so i said, i only code for microcontroller so security is not a concern

**Comment by I-hate-2FA** [comment]
the pr i have reviewed now you can merge

**Comment by asergunov** [comment]
I'd like to keep anode/cathode wording. Also I'd like to have something like 'type: COMMON_ANODE' in case if it will have more than two options one day. 

I'll make PR to your branch when I'll have time so you'll see delta.

**Comment by I-hate-2FA** [comment]
ok, i dont know which one is anode and cathode, so i thought just use common positive and negative would be easy  to understand
and yes there is more then 2 option, theres also charlieplexing which is even more pin saving, it is widely used in modern 7 segment display 6 pins is enough to drive 4 digits

**Comment by asergunov** [comment]
Wkikpedia

> In a simple LED package, typically all of the cathodes (negative terminals) or all of the anodes (positive terminals) of the segment LEDs are connected and brought out to a common pin; this is referred to as a "common cathode" or "common anode" device.

So these are exact words for that. Vxx are flag for voltage rails they can be negative. Ground is a tag to mark common wire of different voltage sources. It's in the middle for audio circuits. Sometimes it's just positive.

**Comment by I-hate-2FA** [comment]
true but in modern micro controller and switching power supply days theres no negative voltage used any more, but i understand your choice to use official term

**Comment by asergunov** [comment]
> there is more then 2 option, theres also charlieplexing which is even more pin saving, it is widely used in modern 7 segment display 6 pins is enough to drive 4 digits

Or it could be display controlled with NPN PNP transistors so all pulses are positive. It's not like "let's implement them all" but "let's keep API stable".

**Comment by asergunov** [comment]
Fixed in #10
